gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,217 @@
+        & ~release_ht.was_split
+        & hl.is_snp(release_ht.alleles[0], release_ht.alleles[1])
+        & hl.is_missing(lcr_intervals.ht()[release_ht.locus])
+    ).key_by("locus")
+
+    # Densify the MT and include only sites defined in the qc_sites ht
+    mt = mt.select_entries(
+        "END", GT=mt.LGT, adj=get_adj_expr(mt.LGT, mt.GQ, mt.DP, mt.LAD)
+    )
+    mt = densify_sites(mt, qc_sites, hl.read_table(last_END_position.path))
+
+    return mt
+
+
+def run_pop_pca(
+    mt: hl.MatrixTable,
+    pop: str,
+    min_af: float = 0.001,
+    min_inbreeding_coeff_threshold: float = -0.25,
+    remove_hard_filtered_samples: bool = True,
+    release=False,
+    high_quality=False,
+    outliers=None,
+    mt_read_if_exists=True,
+    ht_read_if_exists=True,
+    outlier_string="",
+) -> hl.MatrixTable:
+    """
+    Generate the QC MT per specified population and generate PCA info.
+
+    :param mt: The QC MT output by the 'compute_subpop_qc_mt' function
+    :param min_af: Minimum variant allele frequency to retain variant in QC matrix table
+    :param min_inbreeding_coeff_threshold: Minimum site inbreeding coefficient to keep
+    :return: MatrixTable filtered to specific population to use for subpop analysis
+    """
+    # Adjust names for different project subpops that should actually be the same
+    meta_ht = meta.ht()
+    meta_ht = meta_ht.annotate(
+        project_meta=meta_ht.project_meta.annotate(
+            project_subpop=(
+                hl.switch(meta_ht.project_meta.project_subpop.lower())
+                .when("peru", "pel")
+                .when("colombian", "clm")
+                .default(meta_ht.project_meta.project_subpop)
+            )
+        )
+    )
+
+    relateds = pca_related_samples_to_drop.ht()
+    hard_filtered_ht = hard_filtered_samples.ht()
+
+    # Add info to the MT
+    info_ht = get_info(split=False).ht()
+    info_ht = info_ht.annotate(
+        info=info_ht.info.select(
+            # No need for AS_annotations since it's bi-allelic sites only
+            **{x: info_ht.info[x] for x in info_ht.info if not x.startswith("AS_")}
+        )
+    )
+    mt = mt.annotate_rows(info=info_ht[mt.row_key].info)
+
+    # Add sample metadata to the QC MT
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.pop == pop)
+
+    # Apply specified filters
+    if remove_hard_filtered_samples:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(hard_filtered_ht[pop_mt.col_key]))
+    if release:
+        pop_mt = pop_mt.filter_cols(pop_mt.release)
+    if high_quality:
+        pop_mt = pop_mt.filter_cols(pop_mt.high_quality)
+    if outliers is not None:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(outliers[pop_mt.col_key]))
+
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+
+    # Generate a QC MT for the given pop
+    pop_qc_mt = get_qc_mt(
+        pop_mt,
+        min_af=min_af,
```

Comment:
Will need to make sure AF is set to the population we are looking at

Reply:
will this be fine since the mt is filtered to a certain pop and get_qc_mt recomputes AF?

Reply:
Ah, yes, forgot AF is re-computed using this approximation https://github.com/broadinstitute/gnomad_methods/blob/bbb4098b9f67ee35cf5709d8a848102c57048c4d/gnomad/sample_qc/pipeline.py#L55.. In other functions we assume an annotation of `AF` 
and specify the release AF, but with the options above, you are correct, we cant assume that. I think this approximation is fine, we can see how it performs